src/Agent.Plugins/RepositoryPlugin.cs

```diff
@@ -45,7 +45,7 @@ protected RepositoryTask(ISourceProviderFactory sourceProviderFactory)
 
         public Guid Id => Pipelines.PipelineConstants.CheckoutTask.Id;
 
-        public ISourceProviderFactory SourceProviderFactory { get; }
+        public ISourceProviderFactory SourceProviderFactory { get; set; }
```

Comment:
Can this be constructor injected instead of adding a setter?

Reply:
Sorta - not including the setter implicitly makes it readonly and doesn't allow subclasses to change it (even in the constructor), I got around that by removing the setter and calling into the base constructor with the fake object